src/distilabel/llm/mistralai.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2023-present, Argilla, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from functools import cached_property
+from typing import TYPE_CHECKING, Any, Callable, Dict, Generator, List, Optional, Union
+
+from distilabel.llm import LLM
+from distilabel.llm.utils import LLMOutput
+from distilabel.logger import get_logger
+from distilabel.utils.imports import _MISTRALAI_AVAILABLE
+
+if _MISTRALAI_AVAILABLE:
+    from mistralai.client import MistralClient
+    from mistralai.models.chat_completion import ChatMessage
+
+
+if TYPE_CHECKING:
+    from distilabel.tasks.base import Task
+    from distilabel.tasks.prompt import SupportedFormats
+
+logger = get_logger()
+
+
+class MistralAILLM(LLM):
+    def __init__(
+        self,
+        task: "Task",
+        model: str = "mistral-tiny",
```

Comment:
what models did we choose for for example OpenAI? maybe we need to force them to choose a model more specificically

Reply:
perhaps the `mistral-small` is more aligned with `gpt-3.5-turbo` performance? But I'll let you choose.

Reply:
yeah that makes sense